### PR TITLE
Fix menu item hover state in CSS demo page

### PR DESCRIPTION
**Pull request by FlyersPh9**
## Changes

- Remove unused class `.iui-menu-item` from CSS.
- Add `data-iui-actionable='true'` to list items in menu CSS demo page to fix hover issue.

## Testing

Checked hover state manually in both CSS and React demo pages.

## Docs

n/a

### CODE REVIEW

File: packages/itwinui-css/src/menu/mixins.scss [hunk 1]
```diff
@@ -51,7 +51,7 @@ $iui-active-outline: 1px solid var(--iui-color-border-accent);
     margin-block-start: -1px;
   }
 
-  &:where([data-iui-actionable='true'], .iui-menu-item):where(:hover),
+  &:where([data-iui-actionable='true'], .iui-list-item):where(:hover),
```
Comment (mayank99) [inline packages/itwinui-css/src/menu/mixins.scss:54]: I believe this was just leftover from #1295 so it should be removed altogether.

```suggestion
  &:where([data-iui-actionable='true']:hover),
```

By default, `iui-list-item` should *not* have a hover state. See [docs](https://itwinui.bentley.com/docs/list#actionable).

To fix the `menu.html` page, `data-iui-actionable="true"` should be added to all instances of `<div class="iui-list-item">`. See [React implementation](https://github.com/iTwin/iTwinUI/blob/e1045a8ea4e2d36f4b67342e50dfb8278c264217/packages/itwinui-react/src/core/Menu/MenuItem.tsx#L148-L152) for comparison.

Comment (FlyersPh9) [reply packages/itwinui-css/src/menu/mixins.scss]: Ah, this makes sense.  Committed your suggestion and updated the CSS demo page to use `data-iui-actionable='true'`.

File: .changeset/fair-cats-tickle.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-css': patch
+---
+
+Remove unused CSS class.
```
Comment (mayank99) [inline .changeset/fair-cats-tickle.md:5]: I think we can just omit this changeset, it's too small to list out.

**Review by mayank99 — APPROVED** [review]

File: .changeset/fair-cats-tickle.md (hunk 2 above)
Comment (FlyersPh9) [reply .changeset/fair-cats-tickle.md]: Sounds good, deleted.

**Review by smmr-dn — APPROVED** [review]

**Pull request merged**